src/components/tx/NonceForm/index.tsx

```diff
@@ -89,7 +89,7 @@ const NonceForm = ({ name, nonce, recommendedNonce, readonly }: NonceFormProps):
   const label = fieldState.error?.message || nonceWarning || 'Safe transaction nonce'
 
   const onResetNonce = () => {
-    if (recommendedNonce) {
+    if (typeof recommendedNonce === 'number') {
```

Comment:
```suggestion
    if (recommendedNonce >= 0) {
```

Or `!isNaN(recommendedNonce)`.

Reply:
I like `!isNaN(recommendedNonce)`

The validation already checks if the nonce is positive. So I'd say we do not need to check that here.

Reply:
Whilst I agree that both suggestions are cleaner, TypeScript complains that `recommendedNonce` is otherwise `undefined`. I've altered it to `recommendedNonce != null` instead.